src/nativeMain/kotlin/commands/Connect.kt

```diff
@@ -0,0 +1,23 @@
+package commands
+
+import IResultWriter
+import api.IEclairClientBuilder
+import kotlinx.cli.ArgType
+import kotlinx.cli.ExperimentalCli
+import kotlinx.coroutines.runBlocking
+
+@OptIn(ExperimentalCli::class)
+class ConnectCommand(
+    private val resultWriter: IResultWriter,
+    private val eclairClientBuilder: IEclairClientBuilder
+) : BaseCommand(
+    "connect",
+    "Connect to another lightning node. This will perform a connection but no channels will be opened. "
+) {
+    private val uri by argument(ArgType.String, description = "If the uri to the target node is not provided, eclair will use one of the addresses published by the remote peer in its node_announcement messages if it can be found.")
```

Comment:
Why are you using `argument` here instead of the previous `option`? If we change this to:

```suggestion
    private val uri by option(ArgType.String, description = "If the uri to the target node is not provided, eclair will use one of the addresses published by the remote peer in its node_announcement messages if it can be found.").required()
```

The command works and successfully connects to remote nodes.

Reply:
There are 3 options to connect with the another node. These are through `uri`, `manually` and through `nodeId`.
So what i was trying to do is to create multiple subcommands and keep each one of them optional, so that the user can enter any one of them to use the connect command. These subcommands were supposed to look like `connectUri`, `connectManual`, and `connectNodeId`.
And the `.optional()` was only supported for the argument.
The difference between option and argument as stated in the [kotlinx-cli documentation](https://github.com/Kotlin/kotlinx-cli) is:
`Option - command line entity started with some prefix (-/--) and can have value as next entity in command line string.
Argument - command line entity which role is connected only with its position.`
```It's also possible to use arguments in current example.

...
    val input by parser.argument(ArgType.String, description = "Input file")
    val output by parser.argument(ArgType.String, description = "Output file name").optional()
```

If this is a wrong approach(of keeping the subcommands optional),
Then what can be the correct way of doing this?

Reply:
Thanks a lot. I was indeed confused on this.